src/Uno.UI.RuntimeTests/MUX/Utilities/RunOnUIThread.cs

```diff
@@ -173,6 +178,73 @@ public static async Task ExecuteAsync(CoreApplicationView whichView, Action acti
 			}
 		}
 
+		public static async Task ExecuteAsync(CoreApplicationView whichView, Func<Task> task)
+		{
+			Exception exception = null;
+			var dispatcher = whichView.Dispatcher;
+			if (dispatcher.HasThreadAccess)
+			{
+				await task();
+			}
+			else
+			{
+				// We're not on the UI thread, queue the work. Make sure that the action is not run until
+				// the splash screen is dismissed (i.e. that the window content is present).
+				var workComplete = new AutoResetEvent(false);
```

Comment:
Will this work when WASM will be multithreaded?

Reply:
Indeed it didn't! Adjusted the code for Wasm